File: projects/AndroidCustomer/atcustom/src/main/java/com/astir_trotter/atcustom/ui/activity/SplashActivity.java
package com.astir_trotter.atcustom.ui.activity;

import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.TextView;

import com.astir_trotter.atcustom.R;
import com.astir_trotter.atcustom.component.AppInfo;
import com.astir_trotter.atcustom.singleton.Cache;

import java.text.MessageFormat;

public class SplashActivity extends BaseActivity {
    private static final String TAG = SplashActivity.class.getSimpleName();
    private static final long DEFAULT_DELAY_DURATION = 5000;

    private Runnable mDelayedTransit;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (getContentView() != null)
            setContentView(getContentView());
        else if (getContentLayout() != 0)
            setContentView(getContentLayout());
        else {
            setContentView(R.layout.activity_splash);

            AppInfo appInfo = Cache.getInstance().getAppInfo();
            String title = MessageFormat.format("{0} {1}_{2}", appInfo.appName, appInfo.versionName, appInfo.buildNumber);
            ((TextView) findViewById(R.id.app_name)).setText(title);
            ((TextView) findViewById(R.id.app_description)).setText(appInfo.appDescription);
            ((TextView) findViewById(R.id.copyright)).setText(appInfo.copyright);
            ((TextView) findViewById(R.id.org_name)).setText(appInfo.orgName);
        }
    }

    @Override
    protected void onPause() {
        super.onPause();

        if (mDelayedTransit != null) {
            getHandler().removeCallbacks(mDelayedTransit);
        }
    }

    @Override
    protected void onResume() {
        super.onResume();

        fullscreen();

        if (getNextActivityClass() != null) {
            mDelayedTransit = new Runnable() {
                @Override
                public void run() {
                    transit();
                }
            };
            getHandler().postDelayed(mDelayedTransit, getDelayDuration());
        } else
            mDelayedTransit = null;
    }

    @Override
    protected void representTheme() {

    }

    @Override
    protected void representLanguage() {
        if (getContentView() == null &&
                getContentLayout() == 0) {

        }
    }

    private void transit() {
        assert getNextActivityClass() != null;
        transit(getNextActivityClass(), true);
    }

    ////////////////////////////////////////////////////////////////////////////////////////////////
    // getter

    @Nullable
    protected View getContentView() {
        return null;
    }

    @LayoutRes
    protected int getContentLayout() {
        return 0;
    }

    @Nullable
    protected Class<?> getNextActivityClass() {
        return null;
    }

    protected long getDelayDuration() {
        return DEFAULT_DELAY_DURATION;
    }

}

File: projects/AndroidCustomer/atcustom/src/main/java/com/astir_trotter/atcustom/ui/activity/SplashActivity.kt
package com.astir_trotter.atcustom.ui.activity

import android.os.Bundle
import android.support.annotation.LayoutRes
import android.util.Log
import android.view.View
import android.widget.TextView

import com.astir_trotter.atcustom.R
import com.astir_trotter.atcustom.component.AppInfo
import com.astir_trotter.atcustom.singleton.Cache

import java.text.MessageFormat

open class SplashActivity : BaseActivity() {

    private var mDelayedTransit: Runnable? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        Log.d(TAG, "onCreate")

        if (contentView != null)
            setContentView(contentView)
        else if (contentLayout != 0)
            setContentView(contentLayout)
        else {
            setContentView(R.layout.activity_splash)

            val appInfo = Cache.instance.appInfo
            val title = MessageFormat.format("{0} {1}_{2}", appInfo.appName, appInfo.versionName, appInfo.buildNumber)
            (findViewById(R.id.app_name) as TextView).text = title
            (findViewById(R.id.app_description) as TextView).text = appInfo.appDescription
            (findViewById(R.id.copyright) as TextView).text = appInfo.copyright
            (findViewById(R.id.org_name) as TextView).text = appInfo.orgName
        }
    }

    override fun onPause() {
        super.onPause()

        if (mDelayedTransit != null) {
            handler.removeCallbacks(mDelayedTransit)
        }
    }

    override fun onResume() {
        super.onResume()

        fullscreen()

        if (nextActivityClass != null) {
            mDelayedTransit = Runnable { transit() }
            handler.postDelayed(mDelayedTransit, delayDuration)
        } else
            mDelayedTransit = null
    }

    private fun transit() {
        assert(nextActivityClass != null)
        transit(nextActivityClass!!, true)
    }

    ////////////////////////////////////////////////////////////////////////////////////////////////
    // getter

    protected val contentView: View?
        get() = null

    protected val contentLayout: Int
        @LayoutRes
        get() = 0

    protected open val nextActivityClass: Class<*>?
        get() = null

    protected open val delayDuration: Long
        get() = DEFAULT_DELAY_DURATION

    companion object {
        private val TAG = SplashActivity::class.java.simpleName
        private val DEFAULT_DELAY_DURATION: Long = 5000
    }

}
